VUDRMWidevineDemo/app/src/main/java/com/jwplayer/demo/recyclerview/MainActivity.java

```diff
@@ -0,0 +1,232 @@
+		super.onCreate(savedInstanceState);
+		setContentView(R.layout.activity_layout);
+
+		mKeepScreenOnHandler = new KeepScreenOnHandler(getWindow());
+
+		ArrayList<ItemBase> items = new ArrayList<>();
+		items.add(new TextItem("Hi,\nIn this demo we showcase how you can place a JWPlayerView into a RecyclerView and how to implement mutually exclusive playback, so when you press play on one video the other video pauses.\nScroll around and test it out!"));
+
+		// TODO: Enter your content configs -->
+		// Adds item with no DRM
+		items.add(new VideoItem(new PlaylistItem.Builder()
+										.file(INSERT_CONTENT_URL)
+										.title(INSERT_TITLE)
+										.build()));
+		// Adds item with DRM
+		items.add(new VideoItem(new PlaylistItem.Builder()
+										.file(INSERT_CONTENT_URL)
+										.title(INSERT_TITLE)
+										.mediaDrmCallback(new VUDRMCallback(INSERT_TOKEN))
+										.build()));
+
+		RecyclerViewAdapter recyclerViewAdapter = new RecyclerViewAdapter(items);
+
+		RecyclerView recyclerView = findViewById(R.id.recycler_view);
+		RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(this);
+		recyclerView.setLayoutManager(layoutManager);
+		recyclerView.setAdapter(recyclerViewAdapter);
+
+	}
+
+
+	@Override
+	protected void onStop() {
+		super.onStop();
+		for (JWPlayerView player : mPlayers) {
+			player.onStop();
+		}
+	}
+
+	@Override
+	protected void onStart() {
+		super.onStart();
+		for (JWPlayerView player : mPlayers) {
+			player.onStart();
+		}
+	}
+
+	@Override
+	protected void onResume() {
+		// Let JW Player know that the app has returned from the background
+		super.onResume();
+		for (JWPlayerView player : mPlayers) {
+			player.onResume();
+		}
+	}
+
+	@Override
+	protected void onPause() {
+		// Let JW Player know that the app is going to the background
+		super.onPause();
+		for (JWPlayerView player : mPlayers) {
+			player.onPause();
+		}
+	}
+
+	@Override
+	protected void onDestroy() {
+		// Let JW Player know that the app is being destroyed
+		super.onDestroy();
+		for (JWPlayerView player : mPlayers) {
+			player.onDestroy();
+		}
+	}
+
+	@Override
+	public void onFullscreen(FullscreenEvent fullscreenEvent) {
+		ActionBar actionBar = getSupportActionBar();
+		if (actionBar != null) {
+			if (fullscreenEvent.getFullscreen()) {
+				actionBar.hide();
+			} else {
+				actionBar.show();
+			}
+		}
+	}
+
+	@Override
+	public void onPlayerActive(JWPlayerView activePlayer) {
+		mActivePlayer = activePlayer;
+		mKeepScreenOnHandler.addListeners(mActivePlayer);
+
+		for(JWPlayerView player : mPlayers){
+			// If a player was playing, then it was previously set as the active player, pause() and remove listeners
+			if (player.getState() == PlayerState.PLAYING && !player.equals(mActivePlayer)) {
+				player.pause();
+				mKeepScreenOnHandler.removeListeners(player);
+			}
+		}
+	}
+
+
+	public class RecyclerViewAdapter extends RecyclerView.Adapter<RecyclerView.ViewHolder> {
```

Comment:
Any particular reason you chose to include the RecyclerView in your demo? It adds complexity to the demo distracting the developer from required code and optional code.